### PR TITLE
Iframe: calc compat styles once per page load

### PR DESCRIPTION
**Pull request by ellatrix**
## What?

Currently we calculate the compatibility styles for every iframe separately. These styles don't change after load, so they can be reused.

## Why?

First run: -10% (load patterns)
Second run also -10%

Screenshot 2024-01-12 at 11 21 05

### CODE REVIEW

**Review by youknowriad — APPROVED** [review]
Makes sense.

File: packages/block-editor/src/components/iframe/get-compatibility-styles.js [hunk 1]
```diff
@@ -0,0 +1,120 @@
...
+
+			// Don't try to add styles without ID. Styles enqueued via the WP dependency system will always have IDs.
+			if ( ! ownerNode.id ) {
+				return accumulator;
+			}
+
+			function matchFromRules( _cssRules ) {
+				return Array.from( _cssRules ).find(
+					( {
+						selectorText,
```
Comment (jsnajdr) [inline packages/block-editor/src/components/iframe/get-compatibility-styles.js:55]: @ellatrix On WP.com, I'm seeing an error being thrown on this line:
```
Error: Permission denied to access property "selectorText"
```

Screenshot 2024-01-22 at 10 17 06

Any idea what could it be? It happens in Firefox, Chrome seems to be fine.

Comment (ellatrix) [reply packages/block-editor/src/components/iframe/get-compatibility-styles.js:55]: This didn't happen previously? I didn't really change anything here, I just moved the code to a new file.

Comment (jsnajdr) [reply packages/block-editor/src/components/iframe/get-compatibility-styles.js:55]: I don't know, it just started happening randomly on a draft post. And after closing and opening the same post in a new tab the error disappeared again, so I don't really have any steps to reproduce.

But I'm wondering what "Permission denied to access property" even means. It's like the stylesheet object was from a cross-origin iframe, and therefore behind some security boundary.

Comment (ellatrix) [reply packages/block-editor/src/components/iframe/get-compatibility-styles.js:55]: That's why there's a try/catch for accessing `styleSheet.cssRules`, normally it fails there for cross origin stylesheets

**Pull request merged**